shared/actions/teams.tsx

```diff
@@ -64,8 +64,9 @@ const showTeamAfterCreation = (action: TeamsGen.TeamCreatedPayload) => {
   ]
 }
 
-function* joinTeam(_: TypedState, action: TeamsGen.JoinTeamPayload) {
+function* joinTeam(state: TypedState, action: TeamsGen.JoinTeamPayload) {
   const {teamname} = action.payload
+  const teamID = Constants.getTeamID(state, teamname)
```

Comment:
This most likely won't work, we won't have the teamID for this team in the store. I couldn't find anything using the waiting key for this action though, so I think you could just remove it.